src/components/Datepicker/Datepicker.js

```diff
@@ -31,16 +32,120 @@ export class Datepicker extends React.Component {
   constructor(props, context) {
     super(props, context);
     this.prefix = (classes, passThrough) => prefixClasses(this.context.cssPrefix, classes, passThrough);
-    const { open, preselectedDate } = this.props;
+    const { date, initialDate, locale, open, timezone, translations } = this.props;
+
+    moment.locale(locale.toLowerCase());
+    moment.tz.setDefault(timezone);
+
+    if (date && initialDate) {
+      console.warn(
+        '[react-lds] Datepicker:',
+        'You are supplying both `initialDate` & `date`, ignoring `initialDate`.',
+        'The component will work as a controlled component'
+      );
+    }
+
+    let initial;
+    let isValid;
+
+    if (date) {
+      initial = date;
+      isValid = moment(date, iso8601DateFormat, true).isValid();
+    } else if (!date && initialDate) {
+      initial = initialDate;
+      isValid = moment(initialDate, iso8601DateFormat, true).isValid();
+    }
 
     this.state = {
       open,
-      preselectedDate: preselectedDate ? moment(preselectedDate) : null,
-      inputDate: preselectedDate ? moment(preselectedDate).format(salesforceDateFormat).toString() : '',
-      viewedDate: moment(),
+      inputValue: isValid ? moment(initial).format(defaultDateFormat).toString() : '',
+      isInvalid: !isValid,
+      viewedDate: isValid ? moment(initial) : moment(),
+    };
+
+    this.mergedTranslations = {
```

Comment:
Relying on defaultProps here seems dangerous. Abstract the default props into a constant and merge that